test/zipper_test.exs

```diff
@@ -905,4 +906,68 @@ defmodule SourcerorTest.ZipperTest do
       assert "20" == new_zipper |> Z.node() |> Sourceror.to_string()
     end
   end
+
+  describe "at/2" do
+    defp zipper_at_cursor(code_with_cursor) do
+      {position, code} = pop_cursor(code_with_cursor)
+      code |> Sourceror.parse_string!() |> Z.at(position)
+    end
+
+    test "creates a zipper focused on an inner literal at the given position" do
+      assert {:ok, zipper} =
+               zipper_at_cursor("""
+               def foo do
+                 [1, 2, |3, 4, 5]
+               end
+               """)
+
+      assert {:__block__, _, [3]} = zipper |> Z.node()
+
+      assert [
+               {:__block__, _, [1]},
+               {:__block__, _, [2]},
+               {:__block__, _, [3]},
+               {:__block__, _, [4]},
+               {:__block__, _, [5]}
+             ] = zipper |> Z.up() |> Z.node()
+
+      assert {:def, _, _} = zipper |> Z.root()
+    end
+
+    test "creates a zipper focused on a container if position isn't in any children" do
+      assert {:ok, zipper} =
+               zipper_at_cursor("""
+               def foo do
+                 [1|, 2, 3, 4, 5]
+               end
+               """)
+
+      assert [_, _, _, _, _] = zipper |> Z.node()
```

Comment:
This is correct but it make me wonder if we want to push the need to call `Z.up` to get the `:__block__` with the rest of the list literal metadata

Reply:
I'm not 100% sure I'm following. Could you clarify?

Reply:
When you run `Sourceror.parse_string!("[1,2,3,4]")` you get an AST like this:
```elixir
{:__block__,
 [trailing_comments: [], leading_comments: [], closing: [line: 1, column: 9], line: 1, column: 1],
 [
   [
     {:__block__, [trailing_comments: [], leading_comments: [], token: "1", line: 1, column: 2],
      [1]},
     {:__block__, [trailing_comments: [], leading_comments: [], token: "2", line: 1, column: 4],
      [2]},
     {:__block__, [trailing_comments: [], leading_comments: [], token: "3", line: 1, column: 6],
      [3]},
     {:__block__, [trailing_comments: [], leading_comments: [], token: "4", line: 1, column: 8],
      [4]}
   ]
 ]}
```

The literals are being wrapped in a `:__block__` in order to preserve positional information, which makes me wonder if `Zipper.at` here should return the raw list or include the `:__block__` wrapper if any

Reply:
Got it. Right now it's just returning the innermost node that has a known range. Something we could do pretty easily when creating the zipper from the path is match on something like 

```elixir
[{node, [], []}, {:__block__, _ [node]} | ancestors]
```

And in that case, use the block wrapper as the node instead of the literal. This may be more consistent with other literals, since you'll get block nodes wrapping numbers, for instance, since number literals don't have a known range.

Reply:
Yeah that sounds good, let's do that 😄